cmos_noise_map/utils/read_write_utils.py

```diff
@@ -1,124 +1,138 @@
-    amps = gmm_df["Mixture Weights"]
-    return means, var, num_peaks, amps
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jan 24 13:42:45 2023
+
+@author: Prerana Kottapalli
+"""
+from astropy.io import fits
+from glob import glob
+import numpy as np
+import pandas as pd
+import os
+import tempfile
+
+from cmos_noise_map.utils.data_utils import qc_input
+
+def read_bias_frames(path: str, data_ext=0):
+    """
+    A function to open up fits files with memory mapping.
+
+    Parameters
+    ----------
+    path : str
+        DESCRIPTION. The path to your fits files without the .fits at the end
+    data_ext : integer or string, optional
+        DESCRIPTION. The default is 0. This is the extension to the data header data unit (hdu) that is to be processed.
+
+    Returns
+    -------
+    ims : array
+        DESCRIPTION. And array of opened, memory mapped, fits hdus. Not the data itself.
+
+    """
+    # Assumes trimmed and processed bias frames
+    files = glob(path + str("*.fits"), recursive=True)
+    if len(files)==0:
+        images = []
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            file_names = glob(path + str("*.fz"), recursive=True)
+            for filename in file_names:
+                base_filename, file_extension = os.path.splitext(os.path.basename(filename))
+                print(tmpdirname)
+                output_filename = os.path.join(tmpdirname, base_filename)
+                print(output_filename)
+                os.system('funpack -O {0} {1}'.format(output_filename, filename))
+                images.append(fits.open(output_filename, memmap=True, do_not_scale_image_data=True))
+    else:
+        images = [
+            fits.open(f, memmap=True, do_not_scale_image_data=True) for f in files
+        ]  # Doesn't work unless not scaled
+    qc_input(images)
+    images = np.array(images)[:, 0]
+    return images
+
+
+def write_file(data, filename: str, hduname: str = None, data_type="image"):
+    """
+    A function to write an output file depending on which method was used to generate it.
+
+    Parameters
+    ----------
+    data : (NxN array of floats)
+        DESCRIPTION. he data to be written out into the fits file. This is where the readnoise map or the parameter map goes.
+    filename : str
+        DESCRIPTION. Name of the file to be written out. This is without the file ending.
+    hduname : str, optional
+        DESCRIPTION. The name of the hdu in case of writing out a fits file. The default is None.
+    data_type : TYPE, optional
+        DESCRIPTION. Type of data to be written out. This is determined by the program depending on what method was used. The default is "image". If the method used was "param" then the type is "table"
+
+    Returns
+    -------
+    None.
+
+    """
+    if data_type == "image":
+        filename = filename + ".fits"
+        hdr = fits.Header()
+        hdr["NAME"] = hduname
+        hdu = fits.PrimaryHDU(data, header=hdr)
```

Comment:
Just a note for later, we'll probably need to use some of the keywords from the input data to put into the header of the output file. e.g. `SITEID, TELESCOP, INSTRUME` so BANZAI is happy with it